plugins/aep-consent-status/index.ts

```diff
@@ -0,0 +1,75 @@
+/*
+  Copyright 2023 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+import { Event } from '@adobe/griffon-toolkit-common';
+import { ValidationPluginResult } from '../../types/validationPlugin';
+
+(function (events: Event[]): ValidationPluginResult {
+  const { toolkit: kit } = window.griffon;
+  const { sharedState } = kit['aep-mobile'];
+  const stateEvents = kit.match(
+    sharedState.matcher, 
+    events);
+    let consentEvent = null;
+  
+    for (let i = 0; i < stateEvents.length; i++) {
+      const found = stateEvents[i];
+      if (sharedState.getStateOwner(found) === 'com.adobe.edge.consent') {
+         consentEvent = stateEvents[i];
+         //Got the latest consent event, stop searching 
+         break;
+      }
+    }
+    
+    const consentUpdateState = sharedState.getXdm(consentEvent);
+     
+    const valueY = kit.search('consents.collect.val', consentUpdateState) == 'y';
+    const valueN = kit.search('consents.collect.val', consentUpdateState) == 'n';
+    const valueP = kit.search('consents.collect.val', consentUpdateState) == 'p';
```

Comment:
similar comment here. Just store the value once and then compare. Or something like:
```
consentValue = kit.search('consents.collect.val', consentUpdateState);
const valueY = consentValue === 'y';
const valueN = consentValue === 'y';
const valueP = consentValue === 'y';
```

That way the event only has to be searched once.